File: TimberCruisingApp/Screens/PreFieldChecklistScreen.swift
// Phase 7 — pre-field checklist.
//
// Reached from the project dashboard ("Pre-field check" row). Surfaces
// seven readiness checks and a big green/orange/red summary banner so
// the cruiser knows at a glance whether it's safe to drive out.

import SwiftUI
import Models

public struct PreFieldChecklistScreen: View {

    @EnvironmentObject private var environment: AppEnvironment
    @StateObject private var viewModel: PreFieldChecklistViewModel

    public init(project: Project) {
        _viewModel = StateObject(wrappedValue:
            PreFieldChecklistViewModel(project: project))
    }

    public var body: some View {
        List {
            Section {
                summaryBanner
                    .listRowInsets(EdgeInsets())
                    .listRowBackground(Color.clear)
            }
            Section("Checks") {
                ForEach(viewModel.items) { item in
                    row(for: item)
                }
            }
            Section {
                Button {
                    viewModel.runAll()
                } label: {
                    Label("Re-run checks", systemImage: "arrow.clockwise")
                        .frame(maxWidth: .infinity, minHeight: 44)
                }
                .buttonStyle(.borderedProminent)
                .accessibilityIdentifier("prefield.rerun")
            }
        }
        .navigationTitle("Pre-field check")
        #if os(iOS)
        .navigationBarTitleDisplayMode(.inline)
        #endif
        .task {
            viewModel.configure(with: environment)
            viewModel.runAll()
        }
    }

    // MARK: - Summary banner

    @ViewBuilder private var summaryBanner: some View {
        let (title, tint, systemImage) = bannerDescriptor()
        HStack(spacing: 12) {
            Image(systemName: systemImage)
                .font(.system(size: 32, weight: .bold))
                .foregroundStyle(.white)
            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(.title3).bold()
                    .foregroundStyle(.white)
                Text(subtitle)
                    .font(.footnote)
                    .foregroundStyle(.white.opacity(0.9))
            }
            Spacer()
        }
        .padding()
        .background(tint)
        .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        .padding(.horizontal, 16)
        .padding(.vertical, 8)
        .accessibilityIdentifier("prefield.summaryBanner")
    }

    private func bannerDescriptor() -> (String, Color, String) {
        if viewModel.items.isEmpty {
            return ("Running checks…", .gray, "clock")
        }
        if viewModel.items.contains(where: { $0.severity == .fail }) {
            return ("Not ready", .red, "xmark.octagon.fill")
        }
        if viewModel.items.contains(where: { $0.severity == .warn }) {
            return ("Ready with warnings", .orange, "exclamationmark.triangle.fill")
        }
        return ("Ready for field", .green, "checkmark.seal.fill")
    }

    private var subtitle: String {
        let pass = viewModel.items.filter { $0.severity == .pass }.count
        let total = viewModel.items.count
        return "\(pass) of \(total) checks passed"
    }

    // MARK: - Row

    @ViewBuilder private func row(for item: ChecklistItem) -> some View {
        HStack(alignment: .top, spacing: 12) {
            symbol(for: item.severity)
                .accessibilityHidden(true)
            VStack(alignment: .leading, spacing: 4) {
                Text(item.title).font(.headline)
                Text(item.message)
                    .font(.subheadline)
                    .foregroundStyle(.secondary)
                    .fixedSize(horizontal: false, vertical: true)
            }
        }
        .accessibilityElement(children: .combine)
        .accessibilityLabel("\(item.title), \(item.severity.accessibilityWord)")
        .accessibilityValue(item.message)
    }

    @ViewBuilder
    private func symbol(for severity: ChecklistItem.Severity) -> some View {
        switch severity {
        case .pass:
            Image(systemName: "checkmark.circle.fill")
                .foregroundStyle(.green)
        case .warn:
            Image(systemName: "exclamationmark.triangle.fill")
                .foregroundStyle(.orange)
        case .fail:
            Image(systemName: "xmark.octagon.fill")
                .foregroundStyle(.red)
        }
    }
}

private extension ChecklistItem.Severity {
    var accessibilityWord: String {
        switch self {
        case .pass: return "pass"
        case .warn: return "warning"
        case .fail: return "fail"
        }
    }
}

File: android/app/src/main/java/com/hcjeong/forestix/ui/screens/project/PreFieldChecklistScreen.kt
// Port of iOS Screens/PreFieldChecklistScreen.swift.
// Phase 7 — pre-field checklist. Reached from the project dashboard
// ("Pre-field checklist" row). Surfaces seven readiness checks and a big
// green/orange/red summary banner so the cruiser knows at a glance whether
// it's safe to drive out.

package com.hcjeong.forestix.ui.screens.project

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Cancel
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material.icons.filled.Verified
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import com.hcjeong.forestix.LocalAppEnvironment
import com.hcjeong.forestix.data.cruise.Project
import com.hcjeong.forestix.ui.screens.ForestixScaffold
import com.hcjeong.forestix.ui.theme.Forestix
import com.hcjeong.forestix.ui.theme.ForestixProminentButton
import com.hcjeong.forestix.ui.theme.ForestixRadius
import com.hcjeong.forestix.ui.theme.ForestixSpace
import java.util.UUID
import kotlinx.coroutines.launch

@Composable
fun PreFieldChecklistScreen(nav: NavController, projectId: String) {
    val env = LocalAppEnvironment.current
    var project by remember { mutableStateOf<Project?>(null) }
    LaunchedEffect(projectId) {
        project = env.projectRepository.read(UUID.fromString(projectId))
    }
    val loaded = project
    if (loaded == null) {
        ForestixScaffold(nav, title = "Pre-field check") { padding ->
            Box(Modifier.padding(padding).fillMaxSize(), contentAlignment = Alignment.Center) {
                CircularProgressIndicator()
            }
        }
        return
    }
    PreFieldChecklistContent(nav, loaded)
}

@Composable
private fun PreFieldChecklistContent(nav: NavController, project: Project) {
    val env = LocalAppEnvironment.current
    val context = LocalContext.current
    val scope = rememberCoroutineScope()
    val colors = Forestix.colors
    val type = Forestix.type
    val viewModel = remember(project.id) { PreFieldChecklistViewModel(project) }

    val items by viewModel.items.collectAsState()

    // iOS `.task` — configure + run all checks on appear.
    LaunchedEffect(Unit) {
        viewModel.configure(env)
        viewModel.runAll(context)
    }

    ForestixScaffold(nav, title = "Pre-field check") { padding ->
        Column(
            Modifier
                .padding(padding)
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(horizontal = ForestixSpace.md),
            verticalArrangement = Arrangement.spacedBy(ForestixSpace.md),
        ) {
            SummaryBanner(items)

            FormSection(header = "Checks") {
                items.forEach { item ->
                    ChecklistRow(item)
                }
            }

            ForestixProminentButton(
                label = "Re-run checks",
                icon = Icons.Filled.Refresh,        // iOS arrow.clockwise
                modifier = Modifier.fillMaxWidth(),
            ) { scope.launch { viewModel.runAll(context) } }

            Spacer(Modifier.height(ForestixSpace.xl))
        }
    }
}

// MARK: - Summary banner

@Composable
private fun SummaryBanner(items: List<ChecklistItem>) {
    val colors = Forestix.colors
    val type = Forestix.type

    val (title, tint, icon) = bannerDescriptor(items, colors.confidenceOk,
        colors.confidenceWarn, colors.confidenceBad, colors.textSecondary)
    val pass = items.count { it.severity == ChecklistItem.Severity.PASS }
    val subtitle = "$pass of ${items.size} checks passed"

    Row(
        Modifier
            .fillMaxWidth()
            .clip(ForestixRadius.card)
            .background(tint)
            .padding(ForestixSpace.md),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(ForestixSpace.sm),
    ) {
        Icon(icon, contentDescription = null, tint = Color.White,
            modifier = Modifier.size(32.dp))
        Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(title, style = type.title, color = Color.White)
            Text(subtitle, style = type.caption, color = Color.White.copy(alpha = 0.9f))
        }
    }
}

private fun bannerDescriptor(
    items: List<ChecklistItem>,
    ok: Color,
    warn: Color,
    bad: Color,
    gray: Color,
): Triple<String, Color, ImageVector> {
    if (items.isEmpty()) {
        return Triple("Running checks…", gray, Icons.Filled.Schedule)
    }
    if (items.any { it.severity == ChecklistItem.Severity.FAIL }) {
        return Triple("Not ready", bad, Icons.Filled.Cancel)
    }
    if (items.any { it.severity == ChecklistItem.Severity.WARN }) {
        return Triple("Ready with warnings", warn, Icons.Filled.Warning)
    }
    return Triple("Ready for field", ok, Icons.Filled.Verified)
}

// MARK: - Row

@Composable
private fun ChecklistRow(item: ChecklistItem) {
    val colors = Forestix.colors
    val type = Forestix.type
    Row(
        Modifier.fillMaxWidth(),
        verticalAlignment = Alignment.Top,
        horizontalArrangement = Arrangement.spacedBy(ForestixSpace.sm),
    ) {
        val (icon, tint) = when (item.severity) {
            ChecklistItem.Severity.PASS -> Icons.Filled.CheckCircle to colors.confidenceOk
            ChecklistItem.Severity.WARN -> Icons.Filled.Warning to colors.confidenceWarn
            ChecklistItem.Severity.FAIL -> Icons.Filled.Cancel to colors.confidenceBad
        }
        Icon(icon, contentDescription = null, tint = tint, modifier = Modifier.size(20.dp))
        Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text(item.title, style = type.bodyBold, color = colors.textPrimary)
            Text(item.message, style = type.caption, color = colors.textSecondary)
        }
    }
}
